src/js/other-websites/inpage_menu.js

```diff
@@ -0,0 +1,249 @@
+      await browser.runtime.sendMessage({
+        method: "openRelayHomepage",
+      });
+      sendInPageEvent("click", "input-menu-sign-up-btn");
+      iframeCloseRelayInPageMenu();
+    });
+
+    sendInPageEvent("viewed-menu", "unauthenticated-user-input-menu");
+    return;
+  }
+
+  sendInPageEvent("viewed-menu", "authenticated-user-input-menu");
+  
+  // Create "Generate Relay Address" button
+  const generateAliasBtn = document.querySelector(
+    ".fx-relay-menu-generate-alias-btn"
+  );
+
+  generateAliasBtn.textContent = browser.i18n.getMessage(
+    "pageInputIconGenerateNewAlias"
+  );
+
+  // Create "Get unlimited aliases" button
+  const getUnlimitedAliasesBtn = document.querySelector(
+    ".fx-relay-menu-get-unlimited-aliases"
+  );
+  getUnlimitedAliasesBtn.textContent = browser.i18n.getMessage(
+    "popupGetUnlimitedAliases"
+  );
+
+  // If the user has a premium accout, they may create unlimited aliases.
+  const { premium } = await browser.storage.local.get("premium");
+
+  // Create "You have .../.. remaining relay address" message
+  const remainingAliasesSpan = document.querySelector(
+    ".fx-relay-menu-remaining-aliases"
+  );
+
+  const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+  const { maxNumAliases } = await browser.storage.local.get("maxNumAliases");
+
+  const numAliasesRemaining = maxNumAliases - relayAddresses.length;
+
+  // Free user: Set text informing them how many aliases they can create
+  remainingAliasesSpan.textContent = browser.i18n.getMessage(
+    "popupRemainingAliases_2",
+    [numAliasesRemaining, maxNumAliases]
+  );
+
+  // Free user (who once was premium): Set text informing them how they have exceeded the maximum amount of aliases and cannot create any more
+  if (numAliasesRemaining < 0) {
+    remainingAliasesSpan.textContent = browser.i18n.getMessage(
+      "pageFillRelayAddressLimit"
+    );
+  }
+
+  // Premium user: Set text informing them how many aliases they have created so far
+  if (premium) {
+    remainingAliasesSpan.textContent = browser.i18n.getMessage(
+      "popupUnlimitedAliases",
+      [relayAddresses.length]
+    );
+  }
+
+  const maxNumAliasesReached = numAliasesRemaining <= 0;
+
+  // Create "Manage All Aliases" link
+  const relayMenuDashboardLink = document.querySelector(
+    ".fx-relay-menu-dashboard-link"
+  );
+  relayMenuDashboardLink.textContent = browser.i18n.getMessage("ManageAllAliases");
+  relayMenuDashboardLink.href = `${relaySiteOrigin}?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=manage-all-addresses`;
+  relayMenuDashboardLink.addEventListener("click", () => {
+    sendInPageEvent("click", "input-menu-manage-all-aliases-btn");
+  });
+
+  // Create "Get unlimited aliases" link
+  getUnlimitedAliasesBtn.href = `${relaySiteOrigin}/premium?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=get-premium-link`;
+
+  // Restrict tabbing to relay menu elements
+  restrictOrRestorePageTabbing(-1);
```

Comment:
Question: I'm not familiar with this function or how it's supposed to work, but... I can't reach the Relay icon with the keyboard, nor the contents in the menu. Commenting out this function doesn't seem to change anything.

Reply:
Removed in 4344dc8e29ad93376f4af2c58eebb7a81182ed08. This function was necessary to "capture" the focus of the in-page menu injected by the add-on. By switching to the iframe, this is no longer necessary.